design/special-key-space.md

```diff
@@ -78,4 +78,16 @@ ASSERT(
     res2[0].value == LiteralStringRef("London") &&
     res2[1].value == LiteralStringRef("Washington, D.C.")
 );
-```
\ No newline at end of file
+```
+
+## Module
+We introduce this `module` concept after a [discussion](https://forums.foundationdb.org/t/versioning-of-special-key-space/2068) on cross module read on special-key-space. By default, range reads cover more than one module will not be allowed with `special_keys_cross_module_read` errors. In addition, range reads touch no modules will come with `special_keys_no_module_found` errors. The motivation here is to avoid unexpected blocking or errors happen in a wide-scope range read. In particular, you write code `getRange("A", "Z")` when all registered calls between `[A, Z)` happen locally, thus your code does not have any error-handling. However, if in the future, anyone register a new call in `[A, Z)` and sometimes throw errors like `time_out()`, then your original code is broken. The `module` is like a top-level directory where inside the module, calls are homogeneous. So we allow cross range read inside each module by default but cross module reads are forbidden. Right now, there are two modules available to use:
+
+- TRANSACTION : `\xff\xff/transaction/, \xff\xff/transaction0`, all transaction related information like *read_conflict_range*, *write_conflict_range*, *conflicting_keys*.(All happen locally)
+- WORKERINTERFACE : `\xff\xff/worker_interfaces/, \xff\xff/worker_interfaces0`, which is compatible with previous implementation, thus should not be used to add new functions.
+
+In addition, all singleKeyRanges are formatted as modules and cannot be used again. In particular, you should call `get` not `getRange` on these keys. Below are existing ones:
+
+- STATUSJSON : `\xff\xff/status/json`
```

Comment:
If in the future, we want to support `getting one or more specific fields in status json`, how will it affect clients that has used this feature?

Reply:
Since we want to support more specific control for `status/json` in the future, we can use like `\xff\xff/<prefix>/status/<field_name>` for it. And we can support both of them, I mean the original single key `get("\xff\xff/status/json")` and the upgraded one `getRange("\xff\xff/<prefix>/status/<start_key>", "\xff\xff/<prefix>/status/<fend_key>")`.